### PR TITLE
Docs: Tech notes on statistics compuation

### PR DESCRIPTION
**Pull request by TheoRudkiewicz**
New tech notes. Explains the dependencies of the statistics and the updates.

### CODE REVIEW

**Comment by TheoRudkiewicz** [comment]
Extended the technical notes. Normally the compiled version is accessible here:  https://theorudkiewicz.github.io/gromo/tech_notes.html  (it is the correct version if there is a paragraph Statistical quantities).
@streethagore I though you may be interested to review this PR as it explains a bit how the code work. In addition as you are the most new comer you could be more representative of a new user.

File: docs/source/tech_notes.rst [hunk 1]
```diff
@@ -36,6 +40,40 @@ We store those quantities in the `GroMo` module directly. Depending on the situa
...
 When we link only `GrowingModule` modules together, the quantities are stored in the `GrowingModule` module. When we link `GrowingModule` with `AdditionGrowingModule` the quantities are stored in the `AdditionGrowingModule`.  This is due to the fact that we use the `AdditionGrowingModule` when we connect multiple `GrowingModule` together and we do not want to store the same quantities multiple times. Hence we store them in the `AdditionGrowingModule` and we can access them from the `GrowingModule`. Note that we perform the activation after the addition in the `AdditionGrowingModule` so that we can access the pre-activity of the previous layer as the input of the next layer.
 
+
+Statistical quantities
+------------------------
+
+The computation of all statistic has the same structure.
+
+1. Firstly the statistic is created as an instance of `TensorStatistics`.
+2. Then we use our network to compute intermediate quantities. During this forward pass we store the intermediate quantities as explained above and we set the statistics as not updated. This is sed to ensure that we do not update a statistic multiple times with the same data.
```
Comment (streethagore) [inline docs/source/tech_notes.rst:50]:
```suggestion
2. Then we use our network to compute intermediate quantities. During this forward pass we store the intermediate quantities as explained above and we set the statistics as not updated. This is to ensure that we do not update a statistic multiple times with the same data.
```

File: docs/source/tech_notes.rst [hunk 2]
```diff
@@ -36,6 +40,40 @@ We store those quantities in the `GroMo` module directly. Depending on the situa
...
+4. We can then access the statistic by calling it (i.e. using `__call__`).
+
+The previously mentioned statistics (`S local`, `M`, `S growth`, `M prev`, `cross covariance`) are transparently accessible in the `GrowingModule`. However they are not necessary stored in the `GrowingModule`. In the case of a `GrowingModule`,  (`M`, `M prev`, `cross covariance`) are computed in the `GrowingModule`. `S local` is computed either in the previous module if it is an `AdditionGrowingModule` or in the `GrowingModule` if it is a `GrowingModule` (this is due to the fact that all next modules of an addition module require the same `S local`). The computation of `S growth` is left to the previous module in any case (this is due to the fact that in the case of fully-connected layers (`nn.Linear`) the `S growth` is exactly the `S local` of the previous module). (Note that here we talk about the `S growth` used to by the current layer to compute new weights. Indeed, the layer still needs to compute a `S growth` but which is used by the next layer to compute its new weights).
```
Comment (streethagore) [inline docs/source/tech_notes.rst:54]:
```suggestion
The previously mentioned statistics (`S local`, `M`, `S growth`, `M prev`, `cross covariance`) are transparently accessible in the `GrowingModule`. However they are not necessarily stored in the `GrowingModule`. In the case of a `GrowingModule`,  (`M`, `M prev`, `cross covariance`) are computed in the `GrowingModule`. `S local` is computed either in the previous module if it is an `AdditionGrowingModule` or in the `GrowingModule` if it is a `GrowingModule` (this is due to the fact that all next modules of an addition module require the same `S local`). The computation of `S growth` is left to the previous module in any case (this is due to the fact that in the case of fully-connected layers (`nn.Linear`) the `S growth` is exactly the `S local` of the previous module). (Note that here we talk about the `S growth` used by the current layer to compute new weights. Indeed, the layer still needs to compute a `S growth` but which is used by the next layer to compute its new weights).
```

**Review by streethagore — APPROVED** [review]
Approving these modifications.

**Comment by sylvchev** [comment]
Thanks @TheoRudkiewicz

**Pull request merged**